src/main/java/net/earthcomputer/clientcommands/command/MapCommand.java

```diff
@@ -0,0 +1,294 @@
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cmap")
+            .then(literal("export")
+                .executes(ctx -> exportMap(ctx.getSource(), 1))
+                .then(argument("scale", integer(1, 64))
+                    .executes(ctx -> exportMap(ctx.getSource(), getInteger(ctx, "scale"))))));
+    }
+
+    private static int exportMap(FabricClientCommandSource source, int scale) throws CommandSyntaxException {
+        MapItemSavedData data = fromHand(source.getPlayer());
+        if (data != null) {
+            return exportMap(source, new MapInfo[][] { { new MapInfo(data, 0) } }, scale, false);
+        }
+
+        MapInfo[][] worldData = fromWorld(source.getClient().crosshairPickEntity, source.getClient().level, source.getPlayer().getDirection());
+        if (worldData != null) {
+            return exportMap(source, worldData, scale, true);
+        }
+
+        throw NO_HELD_MAP_EXCEPTION.create();
+    }
+
+    private static MapItemSavedData fromHand(LivingEntity entity) {
+        ItemStack item = entity.getMainHandItem();
+        if (item.has(DataComponents.MAP_ID)) {
+            return MapItem.getSavedData(item, entity.level());
+        }
+
+        item = entity.getOffhandItem();
+        if (item.has(DataComponents.MAP_ID)) {
+            return MapItem.getSavedData(item, entity.level());
+        }
+
+        return null;
+    }
+
+    private static MapInfo[][] fromWorld(Entity entity, ClientLevel level, Direction facing) {
+        if (!(entity instanceof ItemFrame frame)) {
+            return null;
+        }
+
+        ItemStack item = frame.getItem();
+        Direction direction = frame.getDirection();
+        if (!item.has(DataComponents.MAP_ID)) {
+            return null;
+        }
+
+        // determine axis directions
+        Direction xAxis = switch (direction) {
+            case NORTH, SOUTH, EAST, WEST -> Direction.DOWN;
+            case UP -> facing.getOpposite();
+            case DOWN -> facing;
+        };
+
+        Direction yAxis = switch (direction) {
+            case NORTH -> Direction.WEST;
+            case EAST -> Direction.NORTH;
+            case SOUTH -> Direction.EAST;
+            case WEST -> Direction.SOUTH;
+            case UP -> Direction.from2DDataValue((xAxis.get2DDataValue() + 3) % 4);
+            case DOWN -> Direction.from2DDataValue((xAxis.get2DDataValue() + 1) % 4);
+        };
+
+        // collect item frames
+        Map<BlockPos, ItemFrame> frames = StreamSupport.stream(level.entitiesForRendering().spliterator(), true)
+            .filter(e -> e instanceof ItemFrame itemFrame && itemFrame.getDirection() == direction)
+            .map(ItemFrame.class::cast)
+            .collect(Collectors.toMap(ItemFrame::getPos, Function.identity()));
```

Comment:
This will crash if there are two item frame entities in the same location. Use either `Collectors.groupingBy` or `Collectors.toMap` with the third `BinaryOperator` argument.